k8s/upgrade-job/src/events/event_recorder.rs

```diff
@@ -274,6 +272,13 @@ impl EventRecorder {
         let _ = self
             .publish_warning(format!("Failed to upgrade: {err}"), "Failed")
             .await
-            .map_err(|error| tracing::error!(%error, "Failed to upgrade {PRODUCT}"));
+            .map_err(|error| error!(%error, "Failed to upgrade {PRODUCT}"));
+    }
+
+    /// Shuts down the event channel which makes the event loop worker exit its loop and return.
+    pub(crate) async fn shutdown_worker(mut self) {
+        let _ = self.event_sender.take();
+
+        self.ack_receiver.recv().await;
```

Comment:
here, a self.task.join().await would have done it I think

Reply:
Ooooh.... great idea